src/main/java/ch/njol/skript/events/EvtCommand.java

```diff
@@ -30,54 +32,60 @@
 import ch.njol.util.StringUtils;
 import ch.njol.util.coll.CollectionUtils;
 
-/**
- * @author Peter Güttinger
- */
-@SuppressWarnings("unchecked")
-public class EvtCommand extends SkriptEvent { // TODO condition to check whether a given command exists, & a conditon to check whether it's a custom skript command
+public class EvtCommand extends SkriptEvent {
 	static {
-		Skript.registerEvent("Command", EvtCommand.class, CollectionUtils.array(PlayerCommandPreprocessEvent.class, ServerCommandEvent.class), "command [%-string%]")
-				.description("Called when a player enters a command (not necessarily a Skript command) but you can check if command is a skript command, see <a href='conditions.html#CondIsSkriptCommand'>Is a Skript command condition</a>.")
-				.examples("on command:", "on command \"/stop\":", "on command \"pm Njol \":")
-				.since("2.0");
+		Skript.registerEvent("Command", EvtCommand.class, CollectionUtils.array(PlayerCommandPreprocessEvent.class, ServerCommandEvent.class), "[1:existing] command [using] [%-strings%]")
```

Comment:
```suggestion
		Skript.registerEvent("Command", EvtCommand.class, CollectionUtils.array(PlayerCommandPreprocessEvent.class, ServerCommandEvent.class), "[:existing] command [using] [%-strings%]")
```
Rather than using 1 for mark, you can use the new tag system which we've been using over marks for understanding easier.

Then in the init method you would do
```diff
- existsOnly = parser.mark == 1;
+ existsOnly  = parseResult.hasTag("existing");
```

`ParseResult parser` should really be called `parseResult` as it's not a reference to the SkriptParser, just the result.

I also don't like the word `existing` it should be something like (-1:s(k|c)ript|1:server) to define a server or skript command. `Existing` is too ambiguous and could mean either.

Then you could use Kleenean for a tri-state and use that for checking and utilize the use of mark numbers.
```diff
private Kleenean check;
^^^^

- existsOnly = parser.mark == 1;
+ check = Kleenean.get(parser.mark); // Unknown = both, -1 = Skript, 1 = Server.
```

```java
if (check == Kleenean.TRUE && event instanceof ServerCommandEvent ...)
```

Reply:
@TheLimeGlass, I'm not sure how I feel about adding `skript/server` vs `existing`
Adding `existing` to the event was done as a means to remove unknown commands from setting off the command event with the fact there's the `on unknown command` event added.

*if anything we could change how this event functions and remove unknown commands by default while adding skript and server but this would end up effecting more servers who don't use paper and relied upon unknown commands in this event*

---

If anything I believe it's better for a server to explicitly define in their code whether they want to support
server or skript commands, while I understand we could argue the same with the `existing` it's just feels more right

```AppleScript

on command:
    command is a server command
    command is a skript command

# vs
on server command:
on skript command:
```

`on server command` just doesn't feel right when compared to `on skript command` which could be an event in it's own